tests/lib/rules/jsx-no-useless-fragment.js

```diff
@@ -63,6 +63,10 @@ ruleTester.run('jsx-no-useless-fragment', rule, {
     {
       code: '<Fooo content={<>eeee ee eeeeeee eeeeeeee</>} />',
       parser: parsers.BABEL_ESLINT
+    },
+    {
+      code: '<>{foos.map(foo => foo)}</>',
```

Comment:
this seems like a useless fragment to me tho? since `foos.map` produces an array, and an array is a valid render value.

Reply:
**In case if `foos` is an array**, the fragment wraps an array (a React node) into a React element. It's not useless if it's intentional (e.g. a function where you pass this treats arrays different than single elements -- requires a single child or sth like that).

https://github.com/yannickcr/eslint-plugin-react/issues/2584#issuecomment-627002431


**But actually, we can't assume what `map` does**, and even if `map` suggests an array, we treat all call expressions the same.
It may return `undefined` and then we have a crash at runtime if render it without wrapping in the fragment.

```js
const foos = { map: () => undefined };
```

Reply:
What crash at runtime? A component that returns `[undefined]` is a valid component.

You're right that it wouldn't be useless if it's being passed somewhere, but since the introduction of fragments it seems like anything that accepts a fragment should also accept an array?

Reply:
> What crash at runtime? A component that returns [undefined] is a valid component.

Yes, but it's not valid if it returns `undefined`. There is no array here.

This doesn't have to be `map`, but even when the function is called "map" we have no guarantee that this is `Array.prototype.map`. A function may return undefined.